lib/controller.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Controller do
+  def main(_args) do
+    run(:welcome_screen)
+  end
+
+  def run(prompt) when prompt != "Q" do
```

Comment:
Fun fact, you could achieve the same thing without the guard clause by moving the next `run` function above this one. That said, I think I like the explicitness of this approach better.

Reply:
moved it above so I don't have to specify what the quit prompt is twice - this way I will only have to change it once when I update how prompts get passed through the controller.